src/app/shared/dovetail/dovetail-audio.ts

```diff
@@ -231,12 +234,22 @@ export class DovetailAudio extends ExtendableAudio {
   private listenerOnError(event: Event) {
     event.stopImmediatePropagation();
     let el = <HTMLMediaElement> event.target;
-    if (this._dovetailOriginalUrl && el.error.code === el.error.MEDIA_ERR_SRC_NOT_SUPPORTED) {
+    let unsupported = el.error.code === el.error.MEDIA_ERR_SRC_NOT_SUPPORTED;
+
+    if (this._dovetailOriginalUrl && unsupported) {
       let url = this._dovetailOriginalUrl;
       this._dovetailOriginalUrl = null;
       this._audio.src = '';
       this.index = -1;
       this.getDovetailDebug(url);
+    } else if (this.index > -1 && unsupported) {
+      this.arrangement.entries[this.index].duration = 0;
+      this.arrangement.entries[this.index].unplayable = true;
+      this.arrangement.duration = undefined;
+      this.$$sendEvent(DURATION_CHANGE);
+      if (this.index > 0) {
```

Comment:
If a non-first segment is unplayable, we can just emit the OnEnded event, and let the default handlers take care of it.